scripts/heatView.py

```diff
@@ -371,23 +401,28 @@ def plot_2D_map(ax, tmin, tmax, xdim, ydim, xwidth, ywidth, title_message, tempe
         #beginning of 3D plotting
         else:
             ax = fig.add_subplot(gs[0], projection='3d')
+            # manually compute z-order
+            ax.computed_zorder = False
+            cache_in_z = 1 if cache_l3 and cache_l3_stacked else 0
+            cache_l3_width = 0 if cache_l3_stacked or not cache_l3 else cache_l3_width
                         #ind = yy + cores_in_y*xx + cores_in_x*cores_in_y*zz
 
+            cores_and_l3_in_y = cores_in_y + cache_l3_width
             #core
             if (arch_type=="3D"):
                 xwidth=(float)(banks_in_x)/cores_in_x
-                ywidth=(float)(banks_in_y)/cores_in_y
+                ywidth=(float)(banks_in_y)/cores_and_l3_in_y
```

Comment:
Do we assume that cache_l3_width is in y-dimension only? Is there any notion of x-dimension also for cache?